linkd-lib/Cargo.toml

```diff
@@ -0,0 +1,42 @@
+[package]
+name    = "linkd-lib"
+version = "0.1.0"
+edition = "2018"
+license = "GPL-3.0-or-later"
+authors = [
+  "xla <self@xla.is>",
+]
+
+[lib]
+doctest = true
+test    = false
+
+[dependencies]
+anyhow              = "1.0"
+base64              = "0.13"
+env_logger          = "0.9"
+futures             = "0.3"
+lazy_static         = "1.4"
+log                 = "0.4"
+nix                 = "0.22"
+signal-hook         = { version = "0.3", default-features = false }
+signal-hook-tokio   = { version = "0.3", features = [ "futures-v0_3" ] }
+structopt           = { version = "0.3", default-features = false }
+thiserror           = "1.0"
+tempfile            = "3.2"
+tokio               = { version = "1.10", default-features = false, features = [ "fs", "io-std", "macros", "process", "rt-multi-thread", "signal" ] }
```

Comment:
Do you need `signal`?

Reply:
Maybe not, or maybe I drop the other signal related code in favour of this. What could go wrong, tokio is production ready software, right?

Reply:
tokio only knows how to handle `CTRL-C` on windows (there's a dedicated API for that). Maybe that's enough, idk, how does one run services there anyway?

Reply:
I concluded until we have more insights into how link is operate on gaming rigs break and ctrl-c is what can be expected.

c792721